packages/extension/src/common/keyNavigationService.js

```diff
@@ -17,88 +16,48 @@ export const validKeys = {
+  if (keyCode === validKeys.l && !insaneMode) return getNextPost(currentElement);
 
-  if (keyCode === validKeys.k && insaneMode) return getAbovePost(currentElement);
+  if (keyCode === validKeys.j) return getBelowPost(currentElement, insaneMode);
 
-  return currentElement;
-}
-
-function triggerBookmark(post) {
-  if (post.ad) return null;
-
-  return post.$emit('bookmark', { post: post.post, bookmarked: !post.post.bookmarked });
-}
-
-function getCurrentPost(posts, current) {
-  if (!current) return null;
+  if (keyCode === validKeys.k) return getAbovePost(currentElement, insaneMode);
 
-  const postOrAdProp = current.post || current.ad;
-
-  return posts.find(article => [article.ad, article.post].indexOf(postOrAdProp) !== -1);
+  return currentElement;
 }
 
-function navigate(keyCode, posts, search, backToMainFeed, { current, insaneMode }) {
-  if (keyCode === validKeys.esc) return backToMainFeed();
+export function navigateDaily(keyCode, posts, hoveredPost, insaneMode) {
+  const foundElement = hoveredPost
+    ? navigateFeed(keyCode, hoveredPost.$el, insaneMode)
+    : getFirstPostOnFeed(insaneMode);
 
-  if (posts.length === 0) return null;
+  const post = posts.find(article => article.$el === foundElement);
 
-  const item = getCurrentPost(posts, current);
+  if (post) return post;
 
-  if (Object.values(validKeys).indexOf(keyCode) === -1) return null;
-
-  if (keyCode === validKeys['/']) return search.enable();
-
-  if (keyCode === validKeys.b && item) return triggerBookmark(item);
-
-  const element = !item
-    ? getTopLeftMostPostEl(posts, insaneMode)
-    : getNewPostEl(keyCode, item.$el, insaneMode);
-
-  const selectedPost = getPostByElement(posts, element);
-
-  if (selectedPost === item) return selectedPost;
-
-  hoverPost(selectedPost);
-
-  return selectedPost;
-}
+  const nextPostAfterEmptyAdElement = getNextPost(foundElement, insaneMode);
 
-function navigateDaily({ keyCode, target }) {
-  if (target instanceof HTMLInputElement || target instanceof HTMLTextAreaElement) {
-    return;
-  }
-  const { daFeedRef, hoveredPost } = store.state.feed;
-  const parent = daFeedRef().$parent;
-  const search = { enable: parent.enableSearch };
-  const options = { insaneMode: store.state.ui.insaneMode, current: hoveredPost };
-  const newHoveredPost = navigate(
-    keyCode,
-    daFeedRef().$refs.posts,
-    search,
-    () => store.dispatch('feed/backToMainFeed'),
-    options,
-  );
-
-  store.commit('feed/setHoveredPost', newHoveredPost);
+  return posts.find(article => article.$el === nextPostAfterEmptyAdElement);
 }
 
-export function enableKeyBindings() {
-  window.addEventListener('keydown', navigateDaily);
+export function bindEvent(event, callback) {
+  window.addEventListener(event, callback);
 }
 
-export function disableKeyBindings() {
-  window.removeEventListener('keydown', navigateDaily);
+export function unbindEvent(event, callback) {
+  window.removeEventListener(event, callback);
 }
```

Comment:
I think these functions are no longer needed.
Other than that it looks great!

Reply:
I didn't realize I wrapped an existing function with the same thing as it does. Lol! 😆 Too much abstraction my friend 🤣